### PR TITLE
Filter chain public key for vaults imported from browser extension

### LINKED ISSUE #3241: [BUG] Seedphrase vault imported from extension shows wrong addresses on Android
**Describe the bug**

A vault that was created via seed phrase import on the browser extension produces different addresses on Android after importing the backup. Regular vaults (created via keygen, not from seed phrase) import correctly.

**To Reproduce**

1. Import a seed phrase into the browser extension (creates a KeyImport vault)
2. Note the addresses shown for each chain on the extension
3. Export the vault backup (.bak file) - use the one received in email
4. Import it on Android
5. Addresses on Android differ from the extension

**Expected behavior**

Addresses should match between the extension and Android after import.

**Screenshots**

Image

**Smartphone (please complete the following information):**
- Device: Any Android device
- OS: Android 8+ (SDK 26+)
- Version: Current main branch

**Additional context**

### Root cause

`ParseVaultFromStringUseCase` does not map `chainPublicKeys` from the protobuf backup. For KeyImport vaults, per-chain public keys are pre-derived from the seed phrase and stored in the `chainPublicKeys` field of the protobuf. These are used directly for address derivation — the root ECDSA key has an empty chain code for KeyImport vaults, which prevents further BIP32 derivation.

Without `chainPublicKeys`, Android falls back to deriving addresses from the root key, which produces incorrect results for KeyImport vaults.

### LINKED ISSUE #3242: [BUG] Importing seed phrase vault with multiple EVM chains shows a misleading error
**Describe the bug**

Importing a seed phrase vault from the browser extension fails with "Incorrect password, try again" when the vault contains multiple chains sharing the same derivation path (e.g. Ethereum + BSC + Polygon, all coinType 60). The actual error is a `SQLiteConstraintException: UNIQUE constraint failed: keyShare.vaultId, keyShare.pubKey` but it is caught generically and displayed as a password error.

This only happens when 2+ chains share the same coinType. A vault with only unique coinType chains (e.g. Bitcoin + Ethereum + Solana) imports fine.

**To Reproduce**

1. Import a seed phrase into the browser extension with 2+ EVM chains selected (e.g. Ethereum + BSC)
2. Export the vault backup (.bak file) - use the one received in email
3. Import it on Android
4. Import fails with "Incorrect password" error

**Expected behavior**

Import should succeed. The duplicate keyshares (same public key from chains sharing a derivation path) should be deduplicated during parsing.

**Screenshots**

Image

**Smartphone (please complete the following information):**
- Device: Any Android device
- OS: Android 8+ (SDK 26+)
- Version: Current main branch

**Additional context**

### Root cause

The browser extension runs a separate DKLS MPC KeyImport session per chain, even when multiple chains share the same derivation path and private key. For example, all EVM chains use coinType 60 (`m/44'/60'/0'/0/0`) and derive the same key. Each MPC session produces different keyshares (random blinding factors) but the same public key. The protobuf backup therefore contains multiple keyshare entries with the same `publicKey`.

Android's `keyShare` table has a composite primary key `(vaultId, pubKey)` with `OnConflictStrategy.ABORT`. Inserting the second keyshare with the same pubKey violates the UNIQUE constraint and throws `SQLiteConstraintException`.

The exception is caught by a generic `catch (e: Exception)` block in `ImportFileViewModel` which shows the password error dialog which is misleading to the user.

### PR DESCRIPTION
**Pull request by rkokhatskyi**
## Description

Fixes two bugs when importing a seed-phrase vault (KeyImport) from the browser extension to Android.

**Bug 1 — Wrong addresses ([#3241](https://github.com/vultisig/vultisig-android/issues/3241)):**
`chainPublicKeys` were never parsed from the protobuf backup. KeyImport vaults need per-chain public keys for address derivation since the root ECDSA key has an empty chain code. Without them all addresses were derived incorrectly. Additionally a filename regex heuristic (`share\d+of\d+`) was overriding `libType` from `KeyImport` to `DKLS` on import, which bypassed all KeyImport-specific logic.

**Bug 2 — Import crash ([#3242](https://github.com/vultisig/vultisig-android/issues/3242)):**
The extension runs a separate MPC KeyImport session per chain, but EVM chains all share coinType 60 (same key). This produces duplicate keyshare entries with the same `publicKey` in the backup. On Android these hit Room's `UNIQUE(vaultId, pubKey)` constraint and throw `SQLiteConstraintException`, which was caught by a generic `catch(Exception)` that showed "Incorrect password".

Both bugs originate from how the extension handles KeyImport (filed as [vultisig-windows#3429](https://github.com/vultisig/vultisig-windows/issues/3429)). This PR fixes the Android side.

### Changes

- **Parse `chainPublicKeys` from protobuf** in `ParseVaultFromStringUseCase` — maps `proto.chainPublicKeys` to `ChainPublicKey` model objects so KeyImport vaults derive correct addresses
- **Deduplicate keyshares by `pubKey`** in `ParseVaultFromStringUseCase` — `.associateBy { it.pubKey }` keeps one keyshare per unique public key, preventing the DB constraint violation
- **Scope filename `libType` override to GG20** in `ImportFileViewModel` — the `share\d+of\d+` heuristic now only applies when `libType == GG20` (old backups), so KeyImport vaults keep their `libType`
- **Catch `SQLiteConstraintException`** in `ImportFileViewModel` — shows "duplicate vault" snackbar instead of "Incorrect password"

- Fixes #3241
- Fixes #3242

## Which feature is affected?

- [ ] Create vault ( Secure / Fast) - Please ensure you created a Secure vault & fast vault
- [x] Import vault
- [ ] Sending  - Please attach a tx link here
- [ ] Swap - Please attach a tx link for the swap here
- [ ] New Chain / Chain related feature  -  Please attach tx link here

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

## Tests

`ParseVaultFromStringUseCaseImplTest` (214 lines):
- Parses basic vault fields from protobuf
- Parses `chainPublicKeys` from protobuf
- Deduplicates keyshares by `pubKey` (keeps last)
- Parses `libType` KeyImport correctly
- Falls back to OldJsonVault decode when protobuf parse fails

`ImportFileViewModelTest` (174 lines):
- KeyImport vault with `share` filename keeps KeyImport `libType`
- GG20 vault with `share` filename gets overridden to DKLS
- DKLS vault without `share` filename keeps DKLS
- `SQLiteConstraintException` on `decryptVaultData` shows duplicate snackbar
- `SQLiteConstraintException` on `saveFileToAppDir` shows duplicate snackbar

## Summary by CodeRabbit

* **New Features**
  * Enhanced duplicate vault detection during import with user-friendly notifications.

* **Bug Fixes**
  * Improved vault import error handling for edge cases.
  * Refined vault type handling during the import process.

* **Tests**
  * Added comprehensive test coverage for vault parsing and import workflows.

### CODE REVIEW

File: app/src/main/java/com/vultisig/wallet/data/usecases/ParseVaultFromStringUseCase.kt [hunk 1]
```diff
@@ -98,7 +113,7 @@ internal class ParseVaultFromStringUseCaseImpl @Inject constructor(
                 }
             )
             json.decodeFromString<OldJsonVaultRoot>(hexToPlainString).vault
-        } catch (e: Exception) {
+        } catch (_: Exception) {
```
[the remark this answers is not in the document]
Comment (rkokhatskyi) [reply app/src/main/java/com/vultisig/wallet/data/usecases/ParseVaultFromStringUseCase.kt]: it is not required. This was always here, just resolved the "code smell"
